osu.Game.Tests/NonVisual/OverridableBindableTest.cs

```diff
@@ -0,0 +1,153 @@
+                maxValue: 11,
+                defaultValue: 5);
+
+            Assert.That(setting.FinalValue.Value, Is.EqualTo(5));
+            Assert.That(setting.FinalValue.Disabled, Is.True);
+
+            setting.BaseValue.Value = 7;
+
+            Assert.That(setting.FinalValue.Value, Is.EqualTo(7));
+            Assert.That(setting.FinalValue.Disabled, Is.True);
+        }
+
+        [Test]
+        public void TestCustomValueSet()
+        {
+            var setting = new OverridableBindable<float>(
+                minValue: 5,
+                maxValue: 15,
+                defaultValue: 6);
+
+            Assert.That(setting.FinalValue.Value, Is.EqualTo(6));
+            Assert.That(setting.FinalValue.Disabled, Is.True);
+
+            setting.CustomValue.Value = 9;
+            setting.HasCustomValue.Value = true;
+
+            Assert.That(setting.FinalValue.Value, Is.EqualTo(9));
+            Assert.That(setting.FinalValue.Disabled, Is.False);
+
+            // change base value to make sure nothing changes in final value.
+            setting.BaseValue.Value = 11;
+
+            Assert.That(setting.FinalValue.Value, Is.EqualTo(9));
+            Assert.That(setting.FinalValue.Disabled, Is.False);
+
+            // change custom value to ensure things are propagating.
+            setting.CustomValue.Value = 6;
+
+            Assert.That(setting.FinalValue.Value, Is.EqualTo(6));
+            Assert.That(setting.FinalValue.Disabled, Is.False);
+
+            // also change externally via FinalValue and ensure propagation inward.
+            setting.FinalValue.Value = 8;
+
+            Assert.That(setting.CustomValue.Value, Is.EqualTo(8));
+
+            // finally, disable custom value to make sure we revert to base and disable changes.
+            setting.HasCustomValue.Value = false;
+
+            Assert.That(setting.FinalValue.Value, Is.EqualTo(11));
+            Assert.That(setting.FinalValue.Disabled, Is.True);
+
+            // re-enable once more for good measure.
+            setting.HasCustomValue.Value = true;
+
+            Assert.That(setting.FinalValue.Value, Is.EqualTo(8));
+            Assert.That(setting.FinalValue.Disabled, Is.False);
+        }
+
+        [Test]
+        public void TestCustomExpectedParsing()
+        {
+            var setting = new OverridableBindable<float>(
+                minValue: 2,
+                maxValue: 8,
+                defaultValue: 2);
+
+            Assert.That(setting.HasCustomValue.Value, Is.False);
+            Assert.That(setting.FinalValue.Value, Is.EqualTo(2));
+
+            parseAndEnsureExpected(7f, 7f);
+            parseAndEnsureExpected(new Bindable<float>(6f), 6f);
+            parseAndEnsureExpected(new OverridableBindable<float>(5f) { HasCustomValue = { Value = true } }, 5f);
+
+            parseAndEnsureExpected(14f, 8f);
+            parseAndEnsureExpected(-5f, 2f);
+
+            void parseAndEnsureExpected(object input, object expected)
```

Comment:
```suggestion
            void parseAndEnsureExpected(object input, float expected)
```

use `object` as sparsely as humanly possible